submit_experiment.py

```diff
@@ -57,7 +57,7 @@ def submit_experiment(
     with ConnectWithPassphrase(hostname) as connection:
         result = connection.run("uname -n", hide=True)
         node = result.stdout.strip()
-        cluster = get_machine_backend(node)
+        cluster = get_machine_backend(node, connection)
```

Comment:
connection is passed only for accesing the username, so i see two options:
1. similarly to `node` we get the username before
2. We pass connection and extract node name inside `get_machine_backend`

Reply:
Option 2 is better imo, implemented